### PR TITLE
csp-3473-enrich-eks-rules

### PR DESCRIPTION
**Pull request by ofiriro3**
We would like the EKS rules to support the new CSP rule format where the meta data is separated from the Rego rules.

- Closes: https://github.com/elastic/security-team/issues/3473

### CODE REVIEW

File: compliance/cis_eks/rules/cis_4_2_1/data.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    Create a PSP as described in the Kubernetes documentation, ensuring that the
+    `.spec.privileged field is omitted or set to false.`
+  impact: >
+    Pods defined with `spec.containers[].securityContext.privileged: true` will
+    not be permitted.
+  default_value: >
+    By default, when you provision an EKS cluster, a pod security policy called
+    eks.privileged
+    is automatically created. The manifest for that policy appears below:
+    ```
```
Comment (ofiriro3) [inline compliance/cis_eks/rules/cis_4_2_1/data.yaml:40]: Problem with viewing this section as code

Comment (ofiriro3) [reply compliance/cis_eks/rules/cis_4_2_1/data.yaml:39]: Solved

File: compliance/cis_eks/rules/cis_3_2_6/data.yaml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    `/etc/systemd/system/kubelet.service.d/10-kubelet-args.conf` on each worker node
+    and add the below parameter at the end of the `KUBELET_ARGS` variable string.
+    ```
+    ----protect-kernel-defaults=true
+    ```
+    **Remediation Method 3:**
+    If using the api configz endpoint consider searching for the status of
+    `"protectKernelDefaults":` by extracting the live configuration from the nodes running
+    kubelet.
+    **See detailed step-by-step configmap procedures in Reconfigure a Node's Kubelet in a Live
```
Comment (jeniawhite) [inline compliance/cis_eks/rules/cis_3_2_6/data.yaml:70]: Same comment with link []()

File: compliance/cis_eks/rules/cis_3_2_7/data.yaml [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    `/etc/systemd/system/kubelet.service.d/10-kubelet-args.conf` on each worker node
+    and add the below parameter at the end of the `KUBELET_ARGS` variable string.
+    ```
+    --make-iptables-util-chains:true
+    ```
+    **Remediation Method 3:**
+    If using the api configz endpoint consider searching for the status of
+    `"makeIPTablesUtilChains": true` by extracting the live configuration from the nodes
+    running kubelet.
+    **See detailed step-by-step configmap procedures in Reconfigure a Node's Kubelet in a Live
```
Comment (jeniawhite) [inline compliance/cis_eks/rules/cis_3_2_7/data.yaml:70]: Link

File: compliance/cis_eks/rules/cis_4_2_1/data.yaml [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    rule: RunAsAny
+    seLinux:
+    rule: RunAsAny	
+    supplementalGroups:
+    rule: RunAsAny
+    volumes:
+    - '*'
+    ```
+  references: |
+    1.https://kubernetes.io/docs/concepts/policy/pod-security-policy/#enabling-pod-security-policies
```
Comment (jeniawhite) [inline compliance/cis_eks/rules/cis_4_2_1/data.yaml:75]: No space between link and number

File: compliance/cis_eks/rules/cis_4_2_4/data.yaml [hunk 5]
```diff
@@ -0,0 +1,47 @@
+metadata:
+  id: 8f13ab3c-ee3e-5c0a-857d-95214a2c3f71
+  name: Minimize the admission of containers wishing to share the hostnetwork namespace (Automated)
```
Comment (jeniawhite) [inline compliance/cis_eks/rules/cis_4_2_4/data.yaml:3]: hostnetwork should be host network

File: compliance/cis_eks/rules/cis_4_2_6/data.yaml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  audit: |
+    Get the set of PSPs with the following command:
+    ```
+    kubectl get psp
+    ```
+    For each PSP, check whether running containers as root is enabled:
+    ```
+    kubectl get psp <name> -o=jsonpath='{.spec.runAsUser.rule}'
+    ```
+    Verify that there is at least one PSP which returns MustRunAsNonRoot or MustRunAs with the
```
Comment (jeniawhite) [inline compliance/cis_eks/rules/cis_4_2_6/data.yaml:28]: MustRunAsNonRoot and MustRunAs should be inline code

File: compliance/cis_eks/rules/cis_5_1_1/data.yaml [hunk 7]
```diff
@@ -0,0 +1,65 @@
+metadata:
+  id: af4cc6d6-dd53-5b13-9a13-8987e5453a16
+  name: Ensure Image Vulnerability Scanning using Amazon ECR imagescanning or a third party provider (Manual)
```
Comment (jeniawhite) [inline compliance/cis_eks/rules/cis_5_1_1/data.yaml:3]: imagescanning should be image scanning

File: compliance/cis_eks/rules/cis_5_3_1/data.yaml [hunk 8]
```diff
@@ -0,0 +1,54 @@
+metadata:
+  id: 7169a9b3-0f98-5b5e-a95b-bfd1d3322817
+  name: Ensure Kubernetes Secrets are encrypted using Customer MasterKeys (CMKs) managed in AWS KMS (Automated)
```
Comment (jeniawhite) [inline compliance/cis_eks/rules/cis_5_3_1/data.yaml:3]: MasterKeys should be Master Keys

File: compliance/cis_eks/rules/cis_5_4_1/data.yaml [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+    designate, such as those that originate from your premises. This helps protect access
+    to your cluster in the case of a vulnerability in the cluster's authentication or
+    authorization mechanism.
+    * Better protection from insider attacks: Authorized networks help protect your
+    cluster from accidental leaks of master certificates from your company's premises.
+    Leaked certificates used from outside Amazon EC2 and outside the authorized IP
+    ranges (for example, from addresses outside your company) are still denied access.
+  audit: |
+    ```
+    Input:
```
Comment (jeniawhite) [inline compliance/cis_eks/rules/cis_5_4_1/data.yaml:30]: Should be outside of code block

File: compliance/cis_eks/rules/cis_5_4_2/data.yaml [hunk 10]
```diff
@@ -0,0 +1,49 @@
+metadata:
+  id: c4faa477-6da2-5220-a678-0200ae5be03f
+  name: Ensure clusters are created with Private Endpoint Enabled andPublic Access Disabled (Manual)
```
Comment (jeniawhite) [inline compliance/cis_eks/rules/cis_5_4_2/data.yaml:3]: andPublic should be and Public

File: compliance/cis_eks/rules/cis_5_4_2/data.yaml [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+    Although Kubernetes API requires an authorized token to perform sensitive actions, a
+    vulnerability could potentially expose the Kubernetes publically with unrestricted access.
+    Additionally, an attacker may be able to identify the current cluster and Kubernetes API
+    version and determine whether it is vulnerable to an attack. Unless required, disabling
+    public endpoint will help prevent such threats, and require the attacker to be on the
+    master's VPC network to perform any attack on the Kubernetes API.
+  audit: ""
+  remediation: ""
+  impact: |
+    Configure the EKS cluster endpoint to be private. See Modifying Cluster
```
Comment (jeniawhite) [inline compliance/cis_eks/rules/cis_5_4_2/data.yaml:25]: Modifying Cluster Endpoint Access should be link

**Review by jeniawhite — CHANGES_REQUESTED** [review]

File: compliance/cis_eks/rules/cis_3_2_2/data.yaml [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+    The output of the above command should return something similar to `--config`
+    `/etc/kubernetes/kubelet/kubelet-config.json` which is the location of the Kubelet
+    config file.
+    Open the Kubelet config file:
+    ```
+    sudo more /etc/kubernetes/kubelet/kubelet-config.json
+    ```
+    Verify that the `"authentication": {"webhook": { "enabled": is set to true.`
+    If the `"authentication": {"mode": { argument` is present check that it is not set to
+    `AlwaysAllow`. If it is not present check that there is a Kubelet config file specified by --
```
[the remark this answers is not in the document]
Comment (ofiriro3) [reply compliance/cis_eks/rules/cis_3_2_2/data.yaml:31]: How did you able to notice it? I am amazed

File: compliance/cis_eks/rules/cis_3_2_2/data.yaml [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+    `/etc/systemd/system/kubelet.service.d/10-kubelet-args.conf` on each worker node
+    and add the below parameter at the end of the `KUBELET_ARGS` variable string.
+    ```
+    --authorization-mode=Webhook
+    ```
+    **Remediation Method 3:**
+    If using the api configz endpoint consider searching for the status of
+    `"authentication.*webhook":{"enabled":true"` by extracting the live configuration from
+    the nodes running kubelet.
+    **See detailed step-by-step configmap procedures in Reconfigure a Node's Kubelet in a Live
```
[the remark this answers is not in the document]
Comment (ofiriro3) [reply compliance/cis_eks/rules/cis_3_2_2/data.yaml:68]: Tried to look for the link myself - couldn't find it

File: compliance/cis_eks/rules/cis_3_2_3/data.yaml [hunk 14]
```diff
@@ -0,0 +1,108 @@
...
+    `/etc/systemd/system/kubelet.service.d/10-kubelet-args.conf` on each worker node
+    and add the below parameter at the end of the `KUBELET_ARGS` variable string.
+    ```
+    --client-ca-file=<path/to/client-ca-file>
+    ```
+    **Remediation Method 3:**
+    If using the api configz endpoint consider searching for the status of
+    `"authentication.*x509":("clientCAFile":"/etc/kubernetes/pki/ca.crt"` by
+    extracting the live configuration from the nodes running kubelet.
+    **See detailed step-by-step configmap procedures in Reconfigure a Node's Kubelet in a Live
```
[the remark this answers is not in the document]
Comment (ofiriro3) [reply compliance/cis_eks/rules/cis_3_2_3/data.yaml:74]: Same as before

File: compliance/cis_eks/rules/cis_3_2_6/data.yaml (hunk 2 above)
Comment (ofiriro3) [reply compliance/cis_eks/rules/cis_3_2_6/data.yaml:70]: Same as before

File: compliance/cis_eks/rules/cis_5_4_2/data.yaml (hunk 11 above)
Comment (ofiriro3) [reply compliance/cis_eks/rules/cis_5_4_2/data.yaml:25]: same

**Review by jeniawhite — APPROVED** [review]

**Pull request merged**